blog/_posts/2020-11-06-hybrid-cloud-with-inlets.md

```diff
@@ -0,0 +1,212 @@
+This model applies for different use-cases:
+perhaps you are in the middle of a digital transformation where some parts of the architecture is deployed on a public cloud, but they still need to integrate with some legacy services
+you have some sensitive data to be kept in a private data center due to data residency regulation
+
+## Tutorial
+You will need an inlets PRO license, start a 14-day free trial.
+
+You'll need:
+- A Kubernetes cluster running on a public cloud (e.g. GKE, AKS, EKS, DOKS, ...)
+- `kubectl`, configured to connect to the cluster
+- A domain and access to your DNS admin panel to create a sub-domain
+- A service, like a database, running locally
+
+As an example, we will connect a WordPress instance running in the cloud with a MySQL server running locally. Still, this solution is perfectly applicable to other databases or services like e.g. an Oracle database, a MinIO cluster or a RabbitMQ service.
+
+![hybrid-mysql-wordpress](/images/2020-11-06-hybrid-cloud-with-inlets/mysql-wordpress.png)
+
+### Create the inlets-pro exit server
+
+Before we start an inlets-pro exit service, create a Kubernetes secret with a token:
+
+``` bash
+kubectl create secret generic inlets-token --from-literal=token=<a random token>
+```
+
+First, start an inlets-pro exit server pod and make it public with a LoadBalancer service:
+
+
+``` yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inlets-pro-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: inlets-pro-server
+  template:
+    metadata:
+      labels:
+        app: inlets-pro-server
+    spec:
+      containers:
+        - name: inlets-pro
+          image: inlets/inlets-pro:0.7.2
+          imagePullPolicy: IfNotPresent
+          command: [ "inlets-pro" ]
+          args:
+            - "server"
+            - "--auto-tls"
+            - "--common-name=inlets.example.com"
+            - "--token-from=/etc/inlets/token"
+          volumeMounts:
+            - name: temp-volume
+              mountPath: /tmp
+            - name: inlets-token
+              mountPath: /etc/inlets
+              readOnly: true   
+      volumes:
+        - name: temp-volume
+          emptyDir: {}        
+        - name: inlets-token
+          secret:
+            secretName: inlets-token
+```
+
+After applying this on the cluster, a exit server pod is available with:
+
+- `auto-tls` enabled, meaning a TLS certificate for the `common-name` is automatically generated
+- the default control port 8123
+- the token available in the previously created secret
+
+Now expose the exit server with a LoadBalancer service:
+
+``` yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: inlets-pro-server
+  labels:
+    app: inlets-pro-server
+spec:
+  type: LoadBalancer
+  ports:
+    - name: control
+      port: 8123
+      targetPort: 8123
+  selector:
+    app: inlets-pro-server
+```
```

Comment:
I would suggest adding a note to say that Kubernetes Ingress can also be used here.